casper/src/main/rholang/BasicWallet.rho

```diff
@@ -0,0 +1,79 @@
+//scalapackage coop.rchain.rholang.wallet
+
+//assumes standard purse API from MakeMint
+contract @"BasicWallet"(@purse, @algorithm, @pk, return) = {
+  new algorithmKnown, cryptoVerifyCh in {
+    match algorithm {
+      "ed25519" => {
+        algorithmKnown!(true) | cryptoVerifyCh!("ed25519Verify")
+      }
+      _ => {
+        algorithmKnown!(false)
+      }
+    } |
+    for(@ak <- algorithmKnown) {
+      match ak {
+        false => { return!([]) }
+        true => {
+          for (cryptoVerify <- cryptoVerifyCh) {
+            new basicWallet, withdraw, verify, nonceCh in {
+              nonceCh!(-1) |
+              contract verify(@nonce, @sig, return) = {
+                for(@prevNonce <- nonceCh) {
+                  match nonce == (prevNonce + 1) {
+                    true => { 
+                      new result in {
+                        cryptoVerify!(nonce.toByteArray(), sig.hexToBytes(), pk.hexToBytes(), *result) |
```

Comment:
I think if we're going to use integers as a nonce, we should have a method that gets a little endian byte array for just the int rather than the process.

Reply:
Also, are we just signing the nonce? That doesn't seem like enough.